com.unity.multiplayer.mlapi/Tests/Editor/InterestTests.cs

```diff
@@ -0,0 +1,344 @@
+            };
+
+            var delegateYes = MakeGameInterestObjectHelper(new Vector3(0.0f, 0.0f, 0.0f), delegateNode);
+            nm.SpawnManager.SpawnNetworkObjectLocally(delegateYes);
+
+            var delegateNo = MakeGameInterestObjectHelper(new Vector3(0.0f, 0.0f, 0.0f), someOtherNode);
+            nm.SpawnManager.SpawnNetworkObjectLocally(delegateNo);
+            Assert.True(delegateNode.SpawnCalled == 1);
+
+            nm.SpawnManager.DespawnObject(delegateYes);
+            nm.SpawnManager.DespawnObject(delegateNo);
+            Assert.True(delegateNode.DespawnCalled == 1);
+        }
+
+        [Test]
+        // Start is called before the first frame update
+        public void AOICustomStorageTests()
+        {
+            var nm = SetUpNetworkingManager();
+
+            var oddsEvensNode = ScriptableObject.CreateInstance<InterestNode>();
+            oddsEvensNode.InterestObjectStorage = new OddsEvensStorage();
+
+            var results = new HashSet<NetworkObject>();
+            var nc = new NetworkClient()
+            {
+                ClientId = 1,
+            };
+            nm.InterestManager.QueryFor(nc, results);
+            var objectsBeforeAdd = results.Count;
+
+            var numNodes = 4;
+            var nodes = new NetworkObject[4];
+            for (var i = 0; i < numNodes; i++)
+            {
+                nodes[i] = MakeGameInterestObjectHelper(new Vector3(0.0f, 0.0f, 0.0f), oddsEvensNode);
+                nodes[i].NetworkObjectId = (ulong)(i + 100);
+                nm.SpawnManager.SpawnNetworkObjectLocally(nodes[i]);
+            }
+
+            nc.PlayerObject = MakeGameInterestObjectHelper(new Vector3(0.0f, 0.0f, 0.0f), null);
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 100;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2);
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 101;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2);
+
+            // now re-number the nodes
+            for (var i = 0; i < numNodes; i++)
+            {
+                nodes[i].NetworkObjectId = (ulong)(2 * i + 101); // 101, 103, 105, 107
+            }
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 100;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2); // should be unchanged
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 101;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2); // should be unchanged
+
+            // now update the nodes
+            for (var i = 0; i < numNodes; i++)
+            {
+                nodes[i].UpdateInterest();
```

Comment:
having to call this, doesn't that defeat the purpose of a node based approach? We'd still need to go over all these objects to check whether they are odds or evens.
Wouldn't it make more sense to add objects to the storage when the object id is changed? To do it as we go to make sure we sort of "cache" that change only once. 
That'd need to be user code, no?

Reply:
objects being considered might not change but conditions of queries might change based on arbitrary reasons (distance, some other state etc.) — so there needs to be a regular update running queries.
having said that, it would be cool to have more control over how often and when node queries are being executed so that we could customize their execution.